File: u and i/ui.js
//Given two sorted arrays, find their union and intersection.
function union(arr1,arr2){
    let union=[]
    let i=0
    let j=0
    while(i<arr1.length && j<arr2.length){
        if(arr1[i]<arr2[j]){
            union.push(arr1[i])
            i++
        }else if(arr1[i]>arr2[j]){
            union.push(arr2[j])
            j++
        }else{
            union.push(arr1[i])
            i++
            j++
        }
    }
    while(i<arr1.length){
        union.push(arr1[i])
        i++
    }
    while(j<arr2.length){
        union.push(arr2[j])
        j++
    }
    return union
}
function intersection(arr1,arr2){
    let intersection=[]
    let i=0
    let j=0
    while(i<arr1.length && j<arr2.length){
        if(arr1[i]<arr2[j]){
            i++
        }else if(arr1[i]>arr2[j]){
            j++
        }else{
            intersection.push(arr1[i])
            i++
            j++
        }
    }
    return intersection
}
console.log(union([1,2,3,4,5,6,7],[2,4,6,8,10,12]))
console.log(intersection([1,2,3,4,5,6,7],[2,4,6,8,10,12]))

File: u and i/ui.py
#Given two sorted arrays, find their union and intersection.

def union(arr1,arr2):

    i = 0
    j = 0
    union = []
    while i < len(arr1) and j < len(arr2):
        if arr1[i] < arr2[j]:
            union.append(arr1[i])
            i += 1
        elif arr1[i] > arr2[j]:
            union.append(arr2[j])
            j += 1
        else:
            union.append(arr1[i])
            i += 1
            j += 1
    while i < len(arr1):
        union.append(arr1[i])
        i += 1
    while j < len(arr2):
        union.append(arr2[j])
        j += 1
    return union
def intersection(arr1,arr2):
    i = 0
    j = 0
    intersection = []
    while i < len(arr1) and j < len(arr2):
        if arr1[i] < arr2[j]:
            i += 1
        elif arr1[i] > arr2[j]:
            j += 1
        else:
            intersection.append(arr1[i])
            i += 1
            j += 1
    return intersection
print(union([1,2,3,4,5],[1,2,3,4,5]))
print(intersection([1,2,3,4,5],[1,2,3,4,5]))